app/models/user.rb

```diff
@@ -1,12 +1,38 @@
 class User < ApplicationRecord
-  include HasPartyAffiliation
+
+  # TODO: when geocoding is enabled
+  # acts_as_mappable :lat_column_name => :latitude, :lng_column_name => :longitude
 
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  rolify
+  rolify after_add: :if_driver_remove_unassigned, after_remove: :make_unassigned
+
+  # TODO: Specs need to be mocked before this is enabled
+  # geocoded_by :full_address
+  # after_create :geocode
+
+  def if_driver_remove_unassigned(role)
+    if self.is_a_driver?
+      self.remove_role(:unassigned_driver)
+    end
+  end
+
+  def make_unassigned(role)
+    unless self.is_a_driver?
+      self.add_role(:unassigned_driver)
+    end
+  end
+
+  def is_a_driver?
+    RideZone.find_roles(nil, self).present?
```

Comment:
This doesn't seem right - won't this find dispatchers in a ride zone too? Should the nil be :driver?